Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/storage/DataDonationPreferences.kt

```diff
@@ -0,0 +1,28 @@
+package de.rki.coronawarnapp.datadonation.storage
+
+import android.content.Context
+import de.rki.coronawarnapp.datadonation.OneTimePassword
+import de.rki.coronawarnapp.util.di.AppContext
+import de.rki.coronawarnapp.util.preferences.clearAndNotify
+import de.rki.coronawarnapp.util.preferences.createFlowPreference
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class DataDonationPreferences @Inject constructor(
```

Comment:
The overall concept is called data donation, with the sub features "event survey" and "privacy preserving analytics".

Maybe we should rename this class as it's "event survey" specific?

Reply:
i thought we could you 1 instance of preferences for data donation. what do you think?

Reply:
As the user gives two different types of consent, one for analytics, and one for surveys, we may run into the situation where the user revokes either one, and we want to clear all data associated with that feature, having the data in different preferences makes that easier then if we have to reset certain key-values individually.

I'm not sure yet though what data we need to safe for analytics, the consent, and? Not sure if @kolyaopahle has started on that yet.

Reply:
good thinking! i'll make 1 instance of preferences for survey only - with according name